mne/connectivity/cfc.py

```diff
@@ -0,0 +1,529 @@
+    filter_data = data.reshape((1, n_times * n_epochs))[0]
+
+    # Filter data into phase modulating freq and amplitude modulated freq.
+    filter_x_fp = band_pass_filter(filter_data, sfreq, l_phase_freq,
+                                   h_phase_freq, method=method, n_jobs=n_jobs)
+    filter_x_fa = band_pass_filter(filter_data, sfreq, l_amp_freq, h_amp_freq,
+                                   method=method, n_jobs=n_jobs)
+
+    # epoch data back again after filtering
+    time_slices = np.arange(0, data.size + n_times, n_times)
+    x_fa = np.zeros(data.shape)
+    x_fp = np.zeros(data.shape)
+    for ind in range(1, n_epochs + 1):
+        x_fa[ind - 1] = filter_x_fa[time_slices[ind - 1]: time_slices[ind]]
+        x_fp[ind - 1] = filter_x_fp[time_slices[ind - 1]: time_slices[ind]]
+
+    if len(x_fp) != len(x_fa) or x_fp.ndim != x_fa.ndim:
+        raise RuntimeError('Phase and amplitude filtered data does'
+                           ' not have same dimensions')
+
+    # Calculate phase series of phase modulating signal
+    phase_series_fp = np.angle(hilbert(x_fp)) + np.pi
+
+    # Calculate amplitude envelope of amplitude modulated signal
+    amp_envelope_fa = np.abs(hilbert(x_fa))
+
+    if surrogates is True:
+        phase_series_fp = shuffle_data(phase_series_fp)
+        amp_envelope_fa = shuffle_data(amp_envelope_fa)
+
+    # Bin the phases
+    bin_size = 2. * np.pi / bin_num  # 360 degrees divided by number of bins
+    phase_bins = np.arange(phase_series_fp.min(), phase_series_fp.max() +
+                           bin_size, bin_size)
+    if len(phase_bins) - 1 != bin_num:
+        raise RuntimeError('Phase bins are incorrect,'
+                           ' please check the number of bins used.')
+
+    # Initialize the arrays
+    digitized = np.zeros(phase_series_fp.shape)
+    amplitude_bin_means = normalized_amplitude = np.zeros((n_epochs, bin_num))
+
+    # Calculate the amplitude distribution for every trial
+    for trial in range(n_epochs):
+        digitized[trial] = np.digitize(phase_series_fp[trial], phase_bins,
+                                       right=False)
+
+        # Calculate mean amplitude at each phase bin
+        amplitude_bin_means[trial] = [amp_envelope_fa[trial][digitized[trial] == i].mean()
+                                      for i in range(1, len(phase_bins))]
+        if np.isnan(np.sum(amplitude_bin_means[trial])):
+            raise ValueError('Encountered nan when calculating mean'
+                             ' amplitude for bins.')
+
+        # Calculate normalized mean amplitude.
+        normalized_amplitude[trial] = (amplitude_bin_means[trial] /
+                                       np.sum(amplitude_bin_means[trial]))
+        assert np.round(normalized_amplitude[trial].sum()) == 1, ('Normalized'
+            ' amplitudes are incorrect')
+
+    return normalized_amplitude, phase_bins
+
+
+def modulation_index(amplitude_distribution):
```

Comment:
Are you meant to pass `normalized_amplitude` (from the above function) to this one? If so, it'd be easier to understand if they were called the same thing or at least clarified in the docstring.